quick-start/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	vault "github.com/hashicorp/vault/api"
+)
+
+const (
+	vaultServerAddr = "http://127.0.0.1:8200"
+	token           = "dev-only-token" // This is the root token for our Vault dev server. Don't do this in production!
+	pathToSecret    = "secret/data/my-secret-password"
+	secretKey       = "password"
+	secretValue     = "Hashi123"
+)
+
+func main() {
+	config := vault.DefaultConfig()
+
+	config.Address = vaultServerAddr
+
+	client, err := vault.NewClient(config)
+	if err != nil {
+		log.Fatalf("unable to initialize Vault client: %v", err)
+	}
+
+	// Authentication
+	client.SetToken(token)
+
+	secretData := map[string]interface{}{
+		"data": map[string]interface{}{
```

Comment:
Should we add a comment with a link to docs explaining the "data" magic? I'm OK leaving it as is as well...

Reply:
I don't think there really is a great straightforward doc about the data magic... at least I don't know of one. So I'm leaning towards leaving it alone so that we're not calling out our own weirdness in their first moments working with Vault. (Though an observant eye will obviously notice it's weird.)